1552_ek_certificate_chain.md

```diff
@@ -0,0 +1,147 @@
+    - [Story 2](#story-2)
+  - [Notes/Constraints/Caveats (optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] Enhancement issue in release milestone, which links to pull request in [keylime/enhancements]
+- [ ] Core members have approved the issue with the label `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place
+- [ ] User-facing documentation has been created in [keylime/keylime-docs]
+
+## Summary
+
+**Issues:**
+* rust-keylime, [#859](https://github.com/keylime/rust-keylime/issues/859)
+* keylime, [#1552](https://github.com/keylime/keylime/issues/1552)
+
+Add support for EK certificates that are signed by intermediate certificates.
+The current setup can only verify EK certificates that are signed by a certificate that
+was added to the `tpm_cert_store`. The current setup does not support the verification of
+a chain.
+
+New [intel chips](https://community.intel.com/t5/Processors/How-to-verify-an-Intel-PTT-endorsement-key-certificate/td-p/1603153/page/2) (starting 11th core gen) are certified using the intel ODCA (On Die Certificate Authority). The certificate chain is as follows:
+
+| tpm_cert_store | TPM | TPM | TPM |
+|:--------------:|:---:|:---:|:---:|
+|Root CA | Intermediate | Intermediate | EK |
+|On Die CSME P_MCC 00001881 Issuing CA| CSME MCC ROM CA | CSME MCC SVN01 Kernel CA | CSME MCC PTT  01SVN |
+
+This is also defined by the [TCG EK specification](https://trustedcomputinggroup.org/wp-content/uploads/TCG-EK-Credential-Profile-V-2.5-R2_published.pdf) (Chapter 2.2.1.5.2)
+
+The current setup only allows to verify the EK cert against the `tpm_cert_store`.
+As the EK certificate is signed by intermediate certificates, which are unique to device,
+it is not possible to verify the EK certificate against the `tpm_cert_store`. The EK
+must be verified against the EK Certificate Chain, till the top level certificate
+from the chain can be verified against the `tpm_cert_store`.
+
+## Motivation
+
+Add support for intel firmware TPMs and possibly other TPMs that provide an EK certificate chain,
+or setups that also depend on a certificate chain that must work offline or without any other
+access to the chain.
+
+### Goals
+
+* Verify EK certificates that are not signed directly by a root ca, but require the verification of a chain.
+
+### Non-Goals
+
+* Implementing a generic interface to provide the content of any NV-Index.
+
+## Proposal
+
+* keylime_agent: update `ekcert` field to use PEM format, so multiple certificates can be stored in the field and a marker exists that shows the start and end of each certificate.
```

Comment:
Just a note: this is an API breaking change as it changes the ABI (it is backwards incompatible). This requires a major API version bump.